lib/src/index.dart

```diff
@@ -414,33 +414,69 @@ class MeiliSearchIndex {
     return await _getTask(http.deleteMethod('/indexes/$uid/documents/$id'));
   }
 
-  /// Delete a selection of documents by given [ids].
-  Future<Task> deleteDocuments(List<Object> ids) async {
-    return await _getTask(
-      http.postMethod(
+  /// Delete a selection of documents by given query.
+  Future<Task> deleteDocuments([DeleteDocumentsQuery? query]) async {
+    Object? deleteBatchData;
+    Object? deleteData;
+    if (query != null) {
+      if (query.ids != null) {
+        // if ids are available, fallback to the delete-batch route
+        deleteBatchData = query.ids;
+      } else {
+        // if ids are NOT available, use the new delete route
+        deleteData = query.toSparseMap();
+      }
+    }
+
+    Future<Response<Map<String, Object?>>> future;
+    if (deleteData != null) {
+      future = http.postMethod(
+        '/indexes/$uid/documents/delete',
+        data: deleteData,
+      );
+    } else {
+      future = http.postMethod(
         '/indexes/$uid/documents/delete-batch',
-        data: ids,
-      ),
-    );
+        data: deleteBatchData,
+      );
+    }
+
+    return await _getTask(future);
   }
 
   /// Return the document in the index by given [id].
-  Future<Map<String, dynamic>?> getDocument(Object id,
-      {List<String> fields = const []}) async {
+  Future<Map<String, dynamic>?> getDocument(
+    Object id, {
+    List<String> fields = const [],
+  }) async {
     final params = DocumentsQuery(fields: fields);
     final response = await http.getMethod<Map<String, dynamic>>(
-        '/indexes/$uid/documents/$id',
-        queryParameters: params.toQuery());
+      '/indexes/$uid/documents/$id',
+      queryParameters: params.toQuery(),
+    );
 
     return response.data;
   }
 
   /// Return a list of all existing documents in the index.
-  Future<Result<Map<String, dynamic>>> getDocuments(
-      {DocumentsQuery? params}) async {
+  Future<Result<Map<String, dynamic>>> getDocuments({
+    DocumentsQuery? params,
+  }) async {
+    if (params != null) {
+      if (params.containsFilter) {
```

Comment:
Can we just have this?
```suggestion
      if (params?.containsFilter) {
```